include/aspect/material_model/interface.h

```diff
@@ -515,9 +515,77 @@ namespace aspect
         {}
     };
 
+    template<int dim>
+    class NamedAdditionalMaterialOutputs: public AdditionalMaterialOutputs<dim>
+    {
+      public:
+        /**
+         * Return a reference to the vector names of the additional
+         * outputs.
+         */
+        const std::vector<std::string> &get_names() const
+        {
+          return names;
+        }
+
+        /**
+         * Given an index as input argument, return a reference the to vector of
+         * values of the additional output with that index.
+         */
+        virtual const std::vector<double> &get_nth_output(const unsigned int idx) const = 0;
+
+        virtual ~NamedAdditionalMaterialOutputs()
+        {}
+
+        virtual void average (const MaterialAveraging::AveragingOperation /*operation*/,
+                              const FullMatrix<double>  &/*projection_matrix*/,
+                              const FullMatrix<double>  &/*expansion_matrix*/)
+        {}
+
+        std::vector<std::string> names;
+    };
 
 
     /**
+     * Additional output fields for the seismic velocities to be added to
+     * the MaterialModel::MaterialModelOutputs structure and filled in the
+     * MaterialModel::Interface::evaluate() function.
+     */
+    template<int dim>
+    class SeismicAdditionalOutputs : public NamedAdditionalMaterialOutputs<dim>
+    {
+      public:
+        SeismicAdditionalOutputs(const unsigned int n_points,
+                                 const unsigned int /*n_comp*/)
+          : vs(n_points, -1.0),
+            vp(n_points, -1.0)
+        {
+          this->names.push_back("seismic_Vs");
+          this->names.push_back("seismic_Vp");
+        }
+
+        virtual const std::vector<double> &get_nth_output(const unsigned int idx) const
+        {
+          switch (idx)
+            {
+              case 0:
+                return vs;
+              case 1:
+                return vp;
+              default:
+                AssertThrow(false, ExcInternalError());
+            }
+          return vs;
+        }
+
+        // seismic s-wave velocity
+        std::vector<double> vs;
```

Comment:
No, then you can not fill them. This is the same as MaterialModelOutputs for example.